src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/health/IndexInfo.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.health
+
+
+import org.elasticsearch.client.core.CountRequest
+import org.elasticsearch.client.core.CountResponse
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import org.springframework.boot.actuate.info.Info
+import org.springframework.boot.actuate.info.InfoContributor
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonersearch.services.IndexStatusService
+import uk.gov.justice.digital.hmpps.prisonersearch.services.SearchClient
+
+
+@Component
+class IndexInfo(private var indexStatusService : IndexStatusService,
+                private val searchClient: SearchClient) : InfoContributor {
+
+  override fun contribute(builder : Info.Builder) {
+    val indexStatus = indexStatusService.getCurrentIndex()
+    builder.withDetail("index-status", indexStatus);
+    builder.withDetail("index-size", mapOf(
+      indexStatus.currentIndex.name to countIndex(indexStatus.currentIndex.indexName).count,
+      indexStatus.currentIndex.otherIndex().name to countIndex(indexStatus.currentIndex.otherIndex().indexName).count
+    ))
+  }
+
+  private fun countIndex(indexName: String): CountResponse {
+    val searchSourceBuilder = SearchSourceBuilder().apply {
```

Comment:
there is a end point in ES that can get this info, from memory it is /<indexName>/_count   but if think you have to use the low level client. Lucy did some code like this 
`      val response = esClient.lowLevelClient.performRequest("get", "/offender/_count", BasicHeader("any", "any"))
      JsonParser.parseString(IOUtils.toString(response.entity.content)).asJsonObject["count"].asInt
`

Reply:
Ah ok, is this faster too?